specutils/analysis/location.py

```diff
@@ -80,16 +85,49 @@ def _centroid_single_region(spectrum, region=None):
     else:
         calc_spectrum = spectrum
 
+    if spectrum.uncertainty is not None:
+        flux_uncert = _convert_uncertainty(calc_spectrum.uncertainty, StdDevUncertainty)
+    else:
+        # dummy value for uncertainties to avoid extra if-statements when applying mask
+        flux_uncert = np.zeros_like(spectrum.flux)
+
     if hasattr(spectrum, 'mask') and spectrum.mask is not None:
         flux = calc_spectrum.flux[~calc_spectrum.mask]
         dispersion = calc_spectrum.spectral_axis[~calc_spectrum.mask].quantity
+        flux_uncert = flux_uncert[~calc_spectrum.mask]
     else:
         flux = calc_spectrum.flux
         dispersion = calc_spectrum.spectral_axis.quantity
 
-    if len(flux.shape) > 1:
+    if len(flux.shape) == 2:
         dispersion = np.tile(dispersion, [flux.shape[0], 1])
+    elif len(flux.shape) > 2:
+        raise ValueError("spectrum must be 1D or 2D")
```

Comment:
I'm a bit confused on this.  Shouldn't it be OK for it to be *any* dimension, and the centroiding is just applied only along the spectral dimension?

Reply:
probably, but then we need an additional argument to control which index is the spectral dimension.  That is probably out of scope here... but you could also argue that the new `ValueError` is out of scope, so I'm happy to remove that and/or open a new ticket to support any number of dimensions.